scipy/special/_dispatch.py

```diff
@@ -0,0 +1,21 @@
+from scipy._lib._array_api import array_namespace
+from . import _ufuncs
+from ._ufuncs import *
+
+__all__ = _ufuncs.__all__
+
+# functools.wraps doesn't work because:
+# 'numpy.ufunc' object has no attribute '__module__'
+def dispatcher(f_name, n_args):
+    def wrapped(*args, **kwargs):
+        xp = array_namespace(*args[:n_args])
+        if xp.__name__.endswith('numpy'):
+            f = getattr(_ufuncs, f_name)
+        else:
+            special = getattr(xp, "special")
+            f = getattr(special, f_name)
```

Comment:
Maybe more like:
```suggestion
        elif xp.__name__.endswith('torch'):
            f = getattr(xp.special, f_name)
        elif xp.__name__.endswith('cupy'):
            import cupyx
            f = getattr(cupyx.scipy.special, f_name)
        elif xp.__name__.endswith('jax'):  # does this work for JAX?
            f = getattr(xp.scipy.special, f_name)
        else:
            # raise error?
            # convert to NumPy array, use SciPy, and convert back?
```

Reply:
`array_api_compat` does not support JAX right now.
It looks like `array_namespace` throws a `TypeError`: https://github.com/data-apis/array-api-compat/blob/546fa3d65dde1cc20c68b2d22241f7977a8c08fa/array_api_compat/common/_helpers.py#L102-L104

Reply:
OK, I wasn't sure and couldn't easily test, so I'll remove that above.

Reply:
Actually, there's no harm in leaving it in. If it is supported in the future, this would probably work.

Reply:
I think the change suggestion here - with explicit handling of CuPy, PyTorch and JAX - is the right idea. Only let's not use `.endswith`, that's too flexible (e.g., a name like `"np_torch" will be caught here, and we don't want that - and that name exists) - let's use the full string name instead. 

And I think it's fine to leave JAX in I'd say, since I'd expect it to be supported in the near future.

Reply:
This if-elif-else block should probably live in a separate function outside this decorator, since in non-ufunc code in `special` we probably want to call a regular function and already will have the `xp = array_namespace(...)` call separately.

Also, it'd be good to not call the decorator `dispatcher`, but something more descriptive. E.g., `support_cupy_torch_jax`.